### DISCUSSION TITLE
Time Series: Legend and Tooltip

### DISCUSSION THREAD

**Discussion by dirkallaert** [grafana/grafana#41104] [Dashboards, Panels & Visualizations]
What would you like to be added:
In the old graph panel, both legend and tooltip could be sorted by value (and null series could be hidden).

Why is this needed:
This is a very useful feature when a time series panel is displaying multiple metrics and you want to quickly see the values of the top metrics.

Sorting legend or tooltip by value is no longer possible with time series. This issue was already raised in several issues. In #30907 it was mentioned "indeed the time series panel does not have legend sorting implemented. Added it to the backlog for time series/graph work: #30564". But both #30907 and #30564 have been closed without this very useful feature being implemented.

**Comment by kvz** [comment]
Sorting by value in the tooldtip allowed our ops team to quickly spot outliers whenever an alert came in, after having switched from Graph (old) -> Timeseries and missing this functionality, it is harder to make sense in such situations. For example:

![image](https://user-images.githubusercontent.com/26752/156001997-c34ffab7-23e4-4199-8d6d-64b9734ec579.png)

compared to previously:

![image](https://user-images.githubusercontent.com/26752/156002714-ccfc4492-994a-4523-a79f-bb0ac641f043.png)

**Comment by zoltanbedi** [comment]
Hi @kvz we added support for that in the new time series panel in 8.4.0 https://github.com/grafana/grafana/pull/43615

**Reply by kvz** [reply]
ohhhh, amazing! thank you!

**Reply by nikobearrr** [reply]
I was looking at issues related to the tooltip and I found this one. Is it on purpose that the default mode is `Single` compared to the old Graph where it was `All`? I would really like if I can make the `All` default (at least for myself)

**Reply by zoltanbedi** [reply]
You can't make All default. Is it on purpose? I don't think so. @dprokop wdyt?

**Reply by zuchka** [reply]
@zoltanbedi I think @nikobearrr is referrring to this: https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/options/builder/tooltip.tsx#L31

changing that to `multi` changes the timeseries panel's default tooltip mode to `All` instead of `Single`

**Comment by kvz** [comment]
I upgraded to 8.4 (it says `Grafana v8.4.2 (NA)` in the help menu) but I don't have the option to sort the tooltip as shown in https://github.com/grafana/grafana/pull/43615 for timeseries graphs :thinking: 

anything else that could be missing?

![image](https://user-images.githubusercontent.com/26752/157106673-4c404353-b9ca-4803-9a7d-230e207eba04.png)

**Reply by zoltanbedi** [reply]
I'm not sure, it should show. I can see it working on https://play.grafana.org

**Reply by kvz** [reply]
This was a deployment issue on our end, sorry for the noise! Works as expected now, thank you
